File: old/BTNode.java
public class BTNode<E> extends Node<E> {
	private BTNode<E> parent;
	private BTNode<E> left;
	private BTNode<E> right;

	public BTNode() {
		this.element = null;
		this.left = null;
		this.right = null;
		this.parent = null;
	}

	public BTNode(E e) {
		this.element = e;
		this.left = null;
		this.right = null;
		this.parent = null;
	}

	public BTNode(BTNode<E> p) {
		this.element = null;
		this.parent = p;
		this.left = null;
		this.right = null;
	}

	public BTNode(E e, BTNode<E> p) {
		this.element = e;
		this.parent = p;
		this.left = null;
		this.right = null;
	}

	public BTNode<E> getLeft() {
		return this.left;
	}

	public void setLeft(BTNode<E> n) {
		this.left = n;
	}

	public BTNode<E> getRight() {
		return this.right;
	}

	public void setRight(BTNode<E> n) {
		this.right = n;
	}

	public boolean isExternal() {
		return((this.left == null) && (this.right == null));
	}

	public boolean isInternal() {
		return (!isExternal());
	}
}

File: c++/oldy/older/BTNode.cpp
#include <iostream>
#include <cstdlib>
#include "BTNode.h"

using std::cout;
using std::endl;
using std::string;

int main() {
	BTNode<string> *root = new BTNode<string>("root");
	BTNode<string> *l = new BTNode<string>("l");
	BTNode<string> *r = new BTNode<string>("r");
	BTNode<string> *ll = new BTNode<string>("ll");
	BTNode<string> *lr = new BTNode<string>("lr");
	BTNode<string> *rl = new BTNode<string>("rl");
	BTNode<string> *rr = new BTNode<string>("rr");

	root->setLeft(*l);
	root->setRight(*r);
	
	cout << *root << endl;
	cout << *(root->left()) << endl;
	cout << *(root->right()) << endl;

	l->setLeft(*ll);
	l->setRight(*lr);

	cout << endl;
	cout << *(l->left()) << endl;
	cout << *(l->right()) << endl;
	
	cout << endl;
	cout << *root->left()->left() << endl;
	cout << *root->left()->right() << endl;
	
	
}
